File: core/src/com/mygdx/game/model/ShotgunProjectile.kt
package com.mygdx.game.model

import com.badlogic.gdx.graphics.Texture
import com.mygdx.game.settings.SHOTGUN_BULLET_SPEED
import com.mygdx.game.settings.STANDARD_PROJECTILE_WIDTH

class ShotgunProjectile(
        x: Float,
        y: Float,
        xSpeed: Float = 0f,
        ySpeed: Float = 0f,
        texture: Texture) : Projectile(
        x,
        y,
        xSpeed,
        ySpeed,
        texture,
        STANDARD_PROJECTILE_WIDTH / 2,
        SHOTGUN_BULLET_SPEED
)

File: core/src/backendServer/models/ShotgunProjectile.js
const Projectile = require('./Projectile');
const ProjectileType = require('./ProjectileType');
const constants = require('../settings/constants');

class ShotgunProjectile extends Projectile {
    constructor(x = 0, y = 0, xSpeed = 0, ySpeed = 0, id, agentId) {
        super(x, y, xSpeed, ySpeed, constants.STANDARD_PROJECTILE_WIDTH / 2, ProjectileType.SHOTGUN,
            constants.SHOTGUN_BULLET_SPEED, constants.SHOTGUN_PROJECTILE_DAMAGE, id, agentId)
    }
}

module.exports = ShotgunProjectile;